pkg/webhook/mock.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package webhook
+
+import "context"
+
+type deliveryEventExistsRes struct {
+	res bool
+	err error
+}
+
+type failureEventsExceedsRetryLimitRes struct {
+	res bool
+	err error
+}
+
+type FakeDatastore struct {
```

Comment:
Could be a follow-up - This says fake but really is a mock. The difference is:

- Fake: A realistic but naive implementation (e.g. in memory storage). Should be unit tested. Often exposed.
- Mock: Not a real implementation. Don't require unit test. Only used in local unit test.

Reply:
Thanks for sharing going to update now.